src/parser/expression.js

```diff
@@ -896,6 +896,14 @@ pp.parseObj = function (isPattern, refShorthandDefaultPos) {
   node.properties = [];
   this.next();
 
+  // `for` keyword begins an object comprehension.
+  if (this.hasPlugin("lightscript") && this.match(tt._for)) {
+    // ...however, `{ for: x }` is a legal JS object.
```

Comment:
Generally speaking I am really against weird edge cases like this that programmers have to remember. I think we can fairly frictionlessly integrate with previously-legal JS at the cost of a lookahead in the parser, which is not that big of a deal. My opinion is we should let people write { for: 4} if they really want to.